server/src/routes/project.ts

```diff
@@ -0,0 +1,67 @@
+import { PrismaClient } from "@prisma/client";
+import express from "express";
+
+export const projectRoutes = express.Router();
+const prisma = new PrismaClient();
+
+projectRoutes.route("/").get(async (req, res) => {
+  try {
+    // todo: find better way of retrieving numeric fields
+    ["id", "expo", "round", "table"].forEach(key => {
+      if (req.query.hasOwnProperty(key))
+        // @ts-ignore
+        req.query[key] = parseInt(req.query[key]);
+    });
+    const projects = await prisma.project.findMany({ where: req.query });
+    res.status(200).send(projects);
+  } catch (err) {
+    console.log(`[ERROR] ${err.message}`);
+    res.status(500).send({ error: true, message: err.message });
+  }
+});
+
+projectRoutes.route("/").post(async (req, res) => {
+  try {
+    // todo: find better way of retreiving required fields
+    const valid = ["name", "description", "githubUrl"]
+      .every(field => req.body.hasOwnProperty(field));
+    if (valid) {
+      const created = await prisma.project.create({ data: req.body });
+      res.status(201).send(created);
+    } else {
+      res.status(400).send("Missing 'name', 'description', or 'githubUrl' fields.");
+    }
```

Comment:
I appreciate this error handling and messaging, but you can just simplify it and just do like `await prisma.project.create({ data: req.body });`. Because our API isn't public and it will only be accessed from the frontend, we don't need this much error handling and messages.